haystack/nodes/answer_generator/openai.py

```diff
@@ -136,13 +136,17 @@ def predict(self, query: str, documents: List[Document], top_k: Optional[int] =
 
         headers = {"Authorization": f"Bearer {self.api_key}", "Content-Type": "application/json"}
         response = requests.request("POST", url, headers=headers, data=json.dumps(payload))
-
         res = json.loads(response.text)
-        generated_answers = [ans["text"] for ans in res["choices"]]
-        answers = self._create_answers(generated_answers, input_docs)
-        result = {"query": query, "answers": answers}
 
-        return result
+        if response.status_code == 200 and "choices" in res:
+            generated_answers = [ans["text"] for ans in res["choices"]]
+            answers = self._create_answers(generated_answers, input_docs)
+            result = {"query": query, "answers": answers}
+            return result
+        else:
+            raise Exception(
+                f"OpenAI returned an error. \nStatus code: {response.status_code}\nResponse body: {response.text}"
+            )
```

Comment:
Minuscule style comment: I like to see exceptions near the conditions that triggered them. So I would just flip the `if-else` here.
```suggestion
        if response.status_code != 200 or "choices" not in res:
            raise Exception(
                f"OpenAI returned an error.\n"
                f"Status code: {response.status_code}\n"
                f"Response body: {response.text}"
            )
            
    generated_answers = [ans["text"] for ans in res["choices"]]
    answers = self._create_answers(generated_answers, input_docs)
    result = {"query": query, "answers": answers}
    return result            
```

Reply:
You can also replace the `Exception` with a custom one. Something like `OpenAIError` (inheriting from `HaystackError`)